pkg/transformer/eth_getTransactionByHash.go

```diff
@@ -19,109 +20,209 @@ func (p *ProxyETHGetTransactionByHash) Method() string {
 	if err != nil {
-		return nil, errors.Wrap(err, "Qtum#DecodeRawTransaction")
+		return nil, errors.WithMessage(err, "couldn't format amount")
 	}
+	ethTx.Value = ethAmount
 
-	ethTx := eth.GetTransactionByHashResponse{
-		Hash:      utils.AddHexPrefix(tx.Txid),
-		BlockHash: utils.AddHexPrefix(tx.Blockhash),
-		Nonce:     "0x01",
-		Value:     ethVal,
+	qtumTxContractInfo, isContractTx, err := qtumDecodedRawTx.ExtractContractInfo()
+	if err != nil {
+		return nil, errors.WithMessage(err, "couldn't extract contract info")
+	}
+	if isContractTx {
+		ethTx.Input = utils.AddHexPrefix(qtum.ZeroUserInput)
+		if qtumTxContractInfo.UserInput == "" {
+			ethTx.Input = utils.AddHexPrefix(qtumTxContractInfo.UserInput)
+		}
+		ethTx.From = utils.AddHexPrefix(qtumTxContractInfo.From)
+		ethTx.To = utils.AddHexPrefix(qtumTxContractInfo.To)
+		ethTx.Gas = utils.AddHexPrefix(qtumTxContractInfo.GasUsed)
+		// TODO: discuss, consider values
+		// - gas limit is needed for `eth_getBlockByHash` request (return as a func result?)
+		// ? ethTx.GasPrice
+		ethTx.GasPrice = "0x0" // temporary solution
+
+		return ethTx, nil
+	}
 
-		// Contract invokation info:
-		// Input,
-		// Gas,
-		// GasPrice,
+	if qtumTx.Generated {
+		ethTx.From = utils.AddHexPrefix(qtum.ZeroAddress)
+	} else {
+		ethTx.From, err = getNonContractTxSenderAddress(p, qtumDecodedRawTx.Vins)
+		if err != nil {
+			return nil, errors.WithMessage(err, "couldn't get non contract transaction sender address")
+		}
+		// TODO: discuss
+		// ? Does func above return incorrect address for graph-node (len is < 40)
+		// ! Temporary solution
+		ethTx.From = utils.AddHexPrefix(qtum.ZeroAddress)
+	}
+	ethTx.To, err = findNonContractTxReceiverAddress(qtumDecodedRawTx.Vouts)
+	if err != nil {
+		return nil, errors.WithMessage(err, "couldn't get non contract transaction receiver address")
+	}
+	// TODO: discuss
+	// ? Does func above return incorrect address for graph-node (len is < 40)
+	// ! Temporary solution
+	ethTx.To = utils.AddHexPrefix(qtum.ZeroAddress)
+
+	// TODO: researching
+	// ! Temporary solution
+	ethTx.Input = utils.AddHexPrefix(qtum.ZeroUserInput)
+	for _, detail := range qtumTx.Details {
+		if detail.Label != "" {
+			ethTx.Input = utils.AddHexPrefix(detail.Label)
+			break
+		}
 	}
 
-	var invokeInfo *qtum.ContractInvokeInfo
+	// TODO: researching
+	// ? Is it correct for non contract transaction
+	ethTx.Gas = "0x0"
+	ethTx.GasPrice = "0x0"
+
+	return ethTx, nil
+}
 
-	// We assume that this tx is a contract invokation (create or call), if we can
-	// find a create or call script.
-	for _, out := range decodedRawTx.Vout {
-		script := strings.Split(out.ScriptPubKey.Asm, " ")
-		finalOp := script[len(script)-1]
+// TODO: discuss
+// ? There are `witness` transactions, that is not acquireable nither via `gettransaction`, nor `getrawtransaction`
+func getRewardTransactionByHash(p *qtum.Qtum, hash string) (*eth.GetTransactionByHashResponse, error) {
```

Comment:
why does this function exist exactly?

Reply:
This function was implemented in order to serve reward transactions, such as coinbase transactions, because those transactions cannot be acquired via gettransaction method, but via getrawtransaction method.
Also implementation of this function helps keep the code a more readable and detailed by splitting the flows, despite some possible duplicates at the moment. However we may, unwrap the code of this function to keep the logic of all cases in one place.
In addition, Qtum may be improved in the future and some changes will be made to this method.

Reply:
Ok that works.